lib/sdf-server/src/server/service/variant_definition/exec_variant_def.rs

```diff
@@ -165,106 +164,430 @@ pub async fn exec_variant_def(
+        variant_spec.root_prop_funcs = exporter
+            .export_root_prop_funcs(
+                &ctx,
+                Some(ctx.visibility().change_set_pk),
+                &previous_variant,
+            )
+            .await?;
+
+        'outer: for old_socket in exporter
+            .export_sockets(
+                &ctx,
+                Some(ctx.visibility().change_set_pk),
+                *previous_variant.id(),
+            )
+            .await?
+        {
+            for socket in &mut variant_spec.sockets {
+                if socket.inputs.is_empty() && socket.name == old_socket.name {
+                    socket.inputs = old_socket.inputs.clone();
+                    if let Some(data) = &mut socket.data {
+                        data.func_unique_id = old_socket
+                            .data
+                            .as_ref()
+                            .and_then(|s| s.func_unique_id.clone());
+                    } else {
+                        socket.data = old_socket.data;
+                    }
+                    continue 'outer;
+                }
+            }
+
+            if let Some(data) = &old_socket.data {
+                if let Some(func_unique_id) = &data.func_unique_id {
+                    if let Ok(func_id) = FuncId::from_str(func_unique_id) {
+                        if let Some(func) = &Func::get_by_id(&ctx, &func_id).await? {
+                            if data.kind == SocketSpecKind::Input {
+                                initial_skips.attribute_skips.push((
+                                    old_socket.name.clone(),
+                                    vec![ImportAttributeSkip::MissingInputSocket {
+                                        name: old_socket.name,
+                                        variant: func.try_into().ok(),
+                                        func_id: *func.id(),
+                                    }],
+                                ));
+                            } else {
+                                initial_skips.attribute_skips.push((
+                                    old_socket.name.clone(),
+                                    vec![ImportAttributeSkip::MissingOutputSocket {
+                                        name: old_socket.name,
+                                        variant: func.try_into().ok(),
+                                        func_id: *func.id(),
+                                    }],
+                                ));
+                            }
+                        }
+                    }
+                }
+            }
+        }
 
-            let schema_variant = SchemaVariant::get_by_id(&ctx, &schema_variant_id)
-                .await?
-                .ok_or(SchemaVariantError::NotFound(schema_variant_id))?;
+        let mut func_ids: Vec<String> = Vec::new();
 
-            let attribute_prototypes = migrate_attribute_functions_to_new_schema_variant(
+        exporter
```

Comment:
What does this do if we throw away the return value?

Reply:
What do you mean by that?